### PR TITLE
Bypass a PHP warning generated by @dns_get_record

### LINKED ISSUE #256: dns_get_record(): DNS Query failed
```php
dns_get_record('test@t12est.co12m', DNS_MX + DNS_A + DNS_AAAA);
```

I got "Warning: dns_get_record(): DNS Query failed".

I suggest add @ to dns_get_record() in https://github.com/egulias/EmailValidator/blob/840d5603eb84cc81a6a0382adac3293e57c1c64c/src/Validation/DNSCheckValidation.php#L121

My environment:
- Windows 10
- PHP 7.3.12 (cli) (built: Nov 19 2019 13:58:02) ( ZTS MSVC15 (Visual C++ 2017) x64 )

### PR DESCRIPTION
**Pull request by alies-dev**
Closes #256 

The idea is to create a custom error handler for `dns_get_record()`, call `dns_get_record` and then restore the original error handler.

This way we can catch PHP warnings (e.g. `Warning: dns_get_record(): DNS Query failed`) and errors: `dns_get_record(): A temporary server error occurred.` [ErrorException].

a new `UnableToGetDNSRecord` extends `NoDNSRecord` for backward compatibility reasons. Note sure about `CODE` const at the exception, I haven't found any rules for it.

## Why @ should be avoided

See https://derickrethans.nl/five-reasons-why-the-shutop-operator-should-be-avoided.html

In short:
 - can produce annoying side effects
 - it's slow

### CODE REVIEW

**Comment by egulias** [comment]
Hi @lptn 
This is a gret PR! A couple of steps before merging:

1. Please do the PR to 3.x & 2.x branches, master is being deprecated
2. Review your test, phpunit is reporting you are comparing `null` to `object`.

Thanks!

**Comment by alies-dev** [comment]
@egulias 
thanks! closing in favor of https://github.com/egulias/EmailValidator/pull/286
